### ISSUE TITLE
Better Syntax Highlighting

### ISSUE THREAD

**Issue by dgarzon** [TypeStrong/atom-typescript#410]
Syntax highlighting seems to be broken, or there may be something wrong with Atom's One Dark Theme. Any feedback on this @basarat? If it is something on our end, I can help with the fix. Attached you will find an example of a .ts file I am working on: 

![screen shot 2015-06-25 at 5 02 16 pm](https://cloud.githubusercontent.com/assets/867990/8365762/69d074b0-1b5d-11e5-86f7-63d3ceb0b041.png)

**Comment by basarat** [comment]
Seems to be missing styles in `one dark` (or us using wrong style names). Compare monokai with one dark:

# One Dark
![image](https://cloud.githubusercontent.com/assets/874898/8368331/32445be4-1bec-11e5-91b7-c95838e0f1c7.png)

# Monokai
![image](https://cloud.githubusercontent.com/assets/874898/8368334/3b349688-1bec-11e5-9283-26bead140265.png)

**Comment by basarat** [comment]
Marked as up for grabs

**Comment by Cu3PO42** [comment]
After experiencing the same bug and finding this issue, I dug around a bit. The keywords that lack highlighting all have the classes `.keyword.operator.ts`. E.g. Monokai sets the highlighting on `.keyword`, you can see the highlight just fine.

Other themes, such as One Dark, Atom Dark and Base16 Tomorrow Dark specifically set the color of `.keyword.operator` back to the normal text color. The classes `.keyword.operator` are also set on normal operators, such as `=`.

If I want these operators highlighted is probably a matter of preference and I get that different themes have different opinions about that. On the other side, I do believe keywords such as `interface`, `declare`, etc. are intended to be highlighted all the time (if you have syntax highlighting turned on).

That would mean atom-typescript is indeed using wrong classes for these keywords. I have never dealt with Atom grammars, but I've had my fair share of fun with Vim's grammars. I'll try and figure out which classes would be more appropriate and send a PR.

**Comment by dgarzon** [comment]
Hey @Cu3PO42! I've been away all day, but after examining how both themes work I came to similar conclusions. I saw your fork and everything is starting to look great! I made one small addition so that `entity.name.module` is colored in a darker shade of orange following on One Dark's color scheme. Here is the code: 

```
.editor, atom-text-editor::shadow {
  .source.ts {
    .meta.function {
      .entity {
        &.name.module {
          color: hsl(29, 54%, 61%);
        }
      }
    }
  }
}
```

I'll keep working on this fix over the weekend. I believe we should add a few more patterns to our grammar file in order to support member variables and functions, and, object keys. Also, given that we are  tweaking One Dark for our needs, maybe my code should be added to that project instead. I saw they have language specific files, so it seems logical. What are your thoughts on this? Hope you have a great night.

**Comment by Cu3PO42** [comment]
@dgarzon I don't really know what Atom's policy is on adding language specifics to themes, but to me that looks fine. Something like that should probably go in One Light as well.

I am starting to dig deeper into TextMate grammars right now to properly match nested types, etc. I only just figured out recursion was available... I must admit I wasted quite some time on thinking about how to decently match a non-regular language with just RegExes.
